build: fix ts-api-guardian not working on windows

devversion opened the pull request:
* Fixes that `ts-api-guardian` does not work on Windows
* Introduces an option that can be used to disable the default Angular project "tag rules".

DevVersion commented on tools/ts-api-guardian/lib/cli.ts:
```diff
@@ -98,7 +107,10 @@ export function parseArguments(input: string[]):
-    }
+    },
+    default: {
+      defaultAngularTags: true,
```
@alexeagle I wasn't too sure about this option. I decided to enable it by default in order to avoid breaking changes and in case someone runs `ts-api-guardian` on his own. I'm also happy to use a better option name.

DevVersion replied:
It might be the best to allow users to specify the "tag rules" somehow through the CLI, but that should be better in a different PR.

alexeagle commented:
Look for Igor's open PR, he is also changing this test a lot

DevVersion commented:
@alexeagle Igor's PR seems to not cover the Windows issues we are experiencing on Material. I've updated the PR to be based on Igor's changes. Please have another look.

Also this PR is blocked now on: #26602

IgorMinar commented on tools/ts-api-guardian/lib/cli.ts:
```diff
@@ -156,25 +176,47 @@ Options:
   process.exit(error ? 1 : 0);
 }
 
+/**
```
the windows support change is independent from the defaultAngularTags change. pleas split this up into two commits and document each in the commit message.

DevVersion replied:
Good point. Done

IgorMinar commented on tools/ts-api-guardian/index.bzl:
```diff
@@ -19,7 +19,15 @@ load("@build_bazel_rules_nodejs//:defs.bzl", "nodejs_binary", "nodejs_test")
+        data = [],
+        strip_export_pattern = ["^__", "^ɵ"],
+        allow_module_identifiers = COMMON_MODULE_IDENTIFIERS,
+        disable_angular_tag_rules = False,
```
please don't use double negatives (negative in name and negative default) - it results in a very confusing code

how about `use_angular_tag_rules = True` ?

IgorMinar commented on tools/ts-api-guardian/index.bzl:
```diff
@@ -39,6 +47,9 @@ def ts_api_guardian_test(name, golden, actual, data = [], strip_export_pattern =
         args += ["--allowModuleIdentifiers", i]
 
+    if disable_angular_tag_rules:
+        args += ["--defaultAngularTags", "false"]
```
we should also rename the command line arg to be the same as attribute name (`useAngularTagRules`)

IgorMinar commented on tools/ts-api-guardian/lib/cli.ts:
```diff
@@ -24,11 +24,28 @@ export function startCli() {
   };
 
+  // Since the API guardian can be also used by other projects, we should provide an option
+  // that can be set to "false" to avoid the Angular project specific tag rules.
```
this is the main reason for your change - can you document it in the commit message? thanks!

IgorMinar reviewed (approved):
the rest looks good to me. please clean up, rebase (my PR landed so you'll need to strip those), and split into two commits.

devversion committed:
build: add option to disable ts-api-guardian default angular tag rules

Since the API guardian can be also used by other projects, we should not set up the default Angular project tag rules unless specified explicitly through a given command option (`useAngularTagRules`)

DevVersion commented:
Closing and re-opening in order to trigger CI again..

IgorMinar commented:
I pulled these changes into #26602

devversion committed:
build: fix ts-api-guardian does not work on windows w/ bazel

* Fixes that the `ts-api-guardian` package does not work on Windows with Bazel. This is because `ts-api-guardian` does not resolve the runfiles through theNodeJS `require` function that properly handles runfiles within Bazel.

DevVersion commented:
@IgorMinar @alexeagle I was going to write a message. The `ts-api-guardian` tests now work on Windows. 

The tests are still somewhat odd because we still use a workaround, but that's better being addressed in a follow-up. Should be good to go now.

IgorMinar commented:
@DevVersion thanks!

IgorMinar commented:
caretaker note: ci flake

alexeagle commented:
/cc @alan-agius4

alan-agius4 commented on tools/ts-api-guardian/lib/cli.ts:
```diff
@@ -156,25 +173,45 @@ Options:
+ * Bazel. We need to use the wrapped NodeJS resolve logic in order to properly handle the given
+ * runfiles files which are only part of the runfile manifest on Windows.
+ */
+function resolveBazelFilePath(fileName: string): string {
```
This is due In case in the runfiles MANIFEST the tree-artifact is a directory it won't be resolved with `rlocation` at the moment. This should eventually be fixed in `bazel 0.20.0`

More info: https://github.com/bazelbuild/bazel/pull/6246

DevVersion replied:
Nice! I didn't bother too much about this because currently with the existing Bazel macro, just a `--out` file can be specified.
